bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/DiscoveryServiceRegistryImpl.java

```diff
@@ -210,13 +212,19 @@ public synchronized void removeDiscoveryListener(DiscoveryListener listener) thr
     }
 
     @Override
-    public synchronized void thingDiscovered(DiscoveryService source, DiscoveryResult result) {
+    public synchronized void thingDiscovered(final DiscoveryService source, final DiscoveryResult result) {
         synchronized (cachedResults) {
             cachedResults.put(source, result);
         }
-        for (DiscoveryListener listener : this.listeners) {
+        for (final DiscoveryListener listener : this.listeners) {
             try {
-                listener.thingDiscovered(source, result);
+                AccessController.doPrivileged(new PrivilegedAction<Void>() {
+                    @Override
+                    public Void run() {
+                        listener.thingDiscovered(source, result);
```

Comment:
I am not an expert on Java security, but doesn't calling all listeners in privilege mode mean that all you have to do is to register such a listener and you can run any malicious code inside it in privileged mode?